src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/routinghelper/TrolleyBusTransportMode.java

```diff
@@ -0,0 +1,38 @@
+package org.openstreetmap.josm.plugins.pt_assistant.actions.routinghelper;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.drew.lang.annotations.NotNull;
+import org.openstreetmap.josm.data.osm.IRelation;
+import org.openstreetmap.josm.data.osm.IWay;
+import org.openstreetmap.josm.data.osm.Node;
+import org.openstreetmap.josm.data.osm.OsmPrimitiveType;
+import org.openstreetmap.josm.data.osm.Relation;
+import org.openstreetmap.josm.data.osm.Way;
+import org.openstreetmap.josm.plugins.pt_assistant.utils.PTIcons;
+import org.openstreetmap.josm.tools.I18n;
+import org.openstreetmap.josm.tools.ImageProvider;
+
+public class TrolleyBusTransportMode extends BusTransportMode {
+    @Override
+    public boolean canTraverseWay(@NotNull final IWay<?> way, @NotNull final WayTraversalDirection direction) {
+        return canTraverseWay(way, direction) && way.hasTag("trolley_wire", "yes");
+    }
+
+    @Override
+    public boolean canBeUsedForRelation(@NotNull final IRelation<?> relation) {
+        return canBeUsedForRelation(relation) && relation.hasTag("route", "trolleybus");
+    }
+
+    @Override
+    public ImageProvider getIcon() {
+        return PTIcons.BUS;
+    } // TODO: find proper icon
```

Comment:
```suggestion
        return super.canTraverseWay(way, direction) && way.hasTag("trolley_wire", "yes");
    }

    @Override
    public boolean canBeUsedForRelation(@NotNull final IRelation<?> relation) {
        return super.canBeUsedForRelation(relation) && relation.hasTag("route", "trolleybus");
    }
```

Reply:
Here you need to use `super`, otherwise the method will call itself in an infinite recursion loop instead of the parent method.

I'd simply leave out the `getIcon()` method until a better icon is found.